Lambda and method type argument inference with non-denotable type

After bumping the checker framework from 2.1.10 to 2.1.14 I realized a change behavior.
I don't know if this is an error, but I don't understand the error at all.

Can you please have a look at?

The annotation KeyForButtom states: "Programmers should rarely write this type."

The error report looks like:

```text
src/Test.java:18: error: [[all, keyfor]:lambda.param.type.incompatible] incompatible parameter types in lambda expression.
        addChangedListener(tmp, variable -> System.out.println(variable));
                                ^
  found   : @UnknownKeyFor Variable<capture#902 of ?[ extends @UnknownKeyFor Object super @UnknownKeyFor Void]>
  required: @UnknownKeyFor Variable<capture#902 of ?[ extends @UnknownKeyFor Object super @KeyForBottom Void]>
```

The project can be found here: https://github.com/maggu2810/checker-framework-demo/tree/master/cf01
So the directory `cf01` of that repository.
You can execute `./compile` to reproduce it easily.

Here is the source code of the file, too:

```Java

public class Test {

    public interface Variable<T> {

        void addChangedListener(VariableChangedListener<T> listener);
    }

    public interface VariableChangedListener<T> {

        void variableChanged(final Variable<T> variable);
    }

    protected <T> void addChangedListener(final Variable<T> variable, final VariableChangedListener<T> listener) {
    }

    public void main(final Variable<?> tmp) {
        addChangedListener(tmp, variable -> System.out.println(variable));
    }

}
```

--- a/checker/tests/nullness/java8inference/Issue1464.java
+++ b/checker/tests/nullness/java8inference/Issue1464.java
@@ -1,0 +1,22 @@
+// Test case for issue 1464
+// https://github.com/typetools/checker-framework/issues/1464
+
+public class Issue1464 {
+
+    public interface Variable<T> {
+
+        void addChangedListener(VariableChangedListener<T> listener);
+    }
+
+    public interface VariableChangedListener<T> {
+
+        void variableChanged(final Variable<T> variable);
+    }
+
+    protected <T> void addChangedListener(
+            final Variable<T> variable, final VariableChangedListener<T> listener) {}
+
+    public void main(final Variable<?> tmp) {
+        addChangedListener(tmp, variable -> System.out.println(variable));
+    }
+}

--- a/framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java
+++ b/framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java
@@ -3560,6 +3560,16 @@ public class AnnotatedTypeFactory implements AnnotationProvider {
                     final AnnotatedTypeMirror newArg;
                     if (types.isSameType(wildcardUbType, correctArgType)) {
                         newArg = wildcardType.getExtendsBound().deepCopy();
+                    } else if (correctArgType.getKind() == TypeKind.TYPEVAR) {
+                        newArg = this.toAnnotatedType(correctArgType, false);
+                        AnnotatedTypeVariable newArgAsTypeVar = (AnnotatedTypeVariable) newArg;
+                        newArgAsTypeVar
+                                .getUpperBound()
+                                .replaceAnnotations(
+                                        wildcardType.getExtendsBound().getAnnotations());
+                        newArgAsTypeVar
+                                .getLowerBound()
+                                .replaceAnnotations(wildcardType.getSuperBound().getAnnotations());
                     } else {
                         newArg = this.toAnnotatedType(correctArgType, false);
                         newArg.replaceAnnotations(wildcardType.getExtendsBound().getAnnotations());

